packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+
+// tell fcl the wallet is ready
+window.parent.postMessage({type: "FCL:FRAME:READY"}, "*")
+```
+
+You will learn fairly fast that almost everything in FCL is optional. The config that FCL sends here is the applications chance to suggest to you, the wallet, what they would like you to send back to them.
+
+An example of that is the OpenID service. The application can request for you, the wallet, to send them the email address of the current user. The application does this through the config highlighted in the example above. The application requesting this information does not mean you need to send it. It's entirely optional for you, the wallet, to do so.
+
+In the config they can also tell you, the wallet, a variety of things about them, the application, such as the name of their application or a url for an icon of their application. You can use these pieces of information to customize your wallets user experience should you desire to do so.
+
+You, the wallet, having a visual distinction from the application, but still a seemless and connected experience is our goal here.
+
+As always, you must never trust anything you recieve from an application. Always do your due-dilligence and be onguard as you, the wallet, are the users first line of defense against potentially malicious applications.
+
+### Authenticate your User 
+
+It's important that you, the wallet, are confident that the user is the user.
+
+Have them provide enough proof to you that you are okay with passing their details back to FCL.
+If we were to use Blocto as an example, this is when the user is logging into Blocto, putting in their email and getting sent an authentication code to their email. Everything Blocto needs to do to be confident in the users identity.
+
+### Once you know who your User is
+
+Once you're confident in the users identity, we can complete the authentication process.
+
+The authentication process is complete once FCL receives back a response, via a post message, that configures FCL with services (more on this concept later) for the current user. This response is extremeley important to FCL. At its core it tells FCL who the user is, and then via included services it tells FCL how the user authenticated, how to request transaction signatures, how to get a personal message signed and the user's email and other details if requested. In the future it may also inlude many more things!
+
+You can kind of think of FCL as a plugin system. But since those plugins exist elsewhere outside of FCL, FCL needs to configured with information on how to communicate with those plugins.
+
+What you are sending back to FCL is everything that it needs to communicate with the plugins that you, the wallet, are supplying.
+Your wallet is like a plugin to FCL, and these details tell FCL how to use you as a plugin.
+
+Here is an example of an authentication resonse:
+
+```javascript
+// IN WALLET AUTHENTICATION FRAME
+window.postMessage({
```

Comment:
Can we replace all the `window.postMessage` stuff with the `WalletUtils` stuff please